src/request.rs

```diff
@@ -1,20 +1,108 @@
-use serde::{Deserialize, Serialize};
+use serde::{
+    de::{self, DeserializeOwned, Visitor},
+    Deserialize, Deserializer, Serialize,
+};
 use serde_repr::{Deserialize_repr, Serialize_repr};
-use std::collections::HashMap;
-use zbus::{fdo::DBusProxy, fdo::Result, Connection};
-use zvariant::{ObjectPath, OwnedValue};
+use std::{collections::HashMap, fmt, marker::PhantomData};
+use zbus::dbus_proxy;
+use zvariant::OwnedValue;
 use zvariant_derive::Type;
 
-/// A typical response returned by the `on_response` signal of a `RequestProxy`.
+/// A typical response returned by the `connect_response` signal of a `RequestProxy`.
 ///
 /// [`RequestProxy`]: ./struct.RequestProxy.html
-pub type Response<T> = std::result::Result<T, ResponseError>;
+#[derive(Debug)]
+pub enum Response<T> {
+    Ok(T),
+    Err(ResponseError),
+}
```

Comment:
is this necessary?

Reply:
Yes, because the responses returned by every request is in the form of (ResponseType, HashMap<String, Value>>), and it's the only sane way to transform it to something that can work across different possible responses

Reply:
Not sure, but maybe Deref to Result could make this a bit simpler to use?

Reply:
> Not sure

The exact issue is that we need to implement `zvariant::Type` & `serde::de::Deserialize` for `Result<T>` which is something you can't do

>  but maybe Deref to Result could make this a bit simpler to use?

Ah, I was looking more into the `Try` trait but it's still unstable.